### DISCUSSION TITLE
Some questions about the rainbow options

### DISCUSSION THREAD

**Discussion by BabakAmini** [izhangzhihao/intellij-rainbow-brackets#2595] [Q&A] [answered]
Thanks for the great plugin. I just started to use the trial version today. Some features are working and some are not. Here is a brief summary of what I have experienced:
  
  - All rainbow features are working in a .py file except for indent guides
  
![image](https://user-images.githubusercontent.com/85762931/219958642-7050021f-8387-4474-8f0b-99ce01cb2022.png)

  - In .js and .jsx files, indent guides are as expected but empty tags remain gray. Also, angel brackets are rainbowified as expected but the element name inside the brackets has the default theme color
  
![image](https://user-images.githubusercontent.com/85762931/219958838-f2be1a2b-9da2-4c37-9f9e-be53027b1db7.png)

![image](https://user-images.githubusercontent.com/85762931/219959594-3a6d0131-9901-4964-99d6-cd31a83e078b.png)

  - In .html files no rainbow colors except for angel brackets
  
![image](https://user-images.githubusercontent.com/85762931/219959814-12c460f9-b2fc-4089-9def-cc7449583206.png)

  

### **My environment:**

![image](https://user-images.githubusercontent.com/85762931/219960002-a5d79dbc-787f-4ca0-aeeb-f0c81aab071a.png)

  PyCharm 2023.1 EAP (Professional Edition)
  Build #PY-231.6471.3, built on January 31, 2023
  Licensed to PyCharm EAP user: *****
  Expiration date: March 2, 2023
  Runtime version: 17.0.6+10-b785.1 amd64
  VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
  Windows 11 10.0
  GC: G1 Young Generation, G1 Old Generation
  Memory: 1024M
  Cores: 8
  Registry:
      ide.intellij.laf.enable.animation=true
      debugger.new.tool.window.layout=true
      documentation.show.toolbar=true
      ide.experimental.ui=true
      ide.balloon.shadow.size=0
  
  Non-Bundled Plugins:
      com.nasller.CodeGlancePro (1.6.0-2023.1)
      monokai-pro (1.10)
      quokka.js (1.0.372)
      com.mroche.JsonHelper (1.2.0)
      com.intellij.ideolog (203.0.30.0)
      com.godwin.json.parser (1.4.1)
      Statistic (4.2.7)
      String Manipulation (9.6.1)
      com.tabnine.TabNine (0.8.13)
      GrepConsole (12.15.211.6693.1)
      NodeJS (231.6471.12)
      izhangzhihao.rainbow.brackets (2023.1.3)
      net.seesharpsoft.intellij.plugins.csv (3.0.3-231)

**Comment by izhangzhihao** [comment, marked as answer]
<img width="999" alt="image" src="https://user-images.githubusercontent.com/12044174/219984447-d45bbf5a-f8f1-41e0-a127-fd5eb58f01a8.png">

**Comment by izhangzhihao** [comment]
Could you turn on these two options?

For the .jsx files, empty tags didn't have any indent guide lines, so it won't ranibowify any indent guide lines.

**Comment by BabakAmini** [comment]
I enabled those options. Now Python and HTML files are rendering as expected, however, I wish to have rainbow indent guides without Python keywords.

For .js files, I didn't select the **Do NOT rainbowify brackets without content** option and was expecting to have them as a non-gray color.

![image](https://user-images.githubusercontent.com/85762931/220151424-e472fc54-8def-441b-923d-8f471ebb82ed.png)

Also, I noticed that all the indent guides for all file types are dimmed now without me making any changes in my editor's configurations (consider comparing bellow screenshot with the third picture in my original post):

![image](https://user-images.githubusercontent.com/85762931/220152443-925794ee-43e5-431e-8eb2-dff3c0b233e1.png)

I'm also wondering what the option **Cycle count on all brackets** does.

**Comment by izhangzhihao** [comment]
- have rainbow indent guides without Python keywords? that's something I could try later.
- all the indent guides for all file types are dimmed? yes, indent guidelines will be highlighted only when you click it.
- Cycle count on all brackets? take a look here: https://github.com/izhangzhihao/intellij-rainbow-brackets/issues/391

**Reply by BabakAmini** [reply]
> * all the indent guides for all file types are dimmed? yes, indent guidelines will be highlighted only when you click it.

When I first installed the plugin, all top level guide lines were highlighted and that was awesome:

![01](https://user-images.githubusercontent.com/85762931/220340622-6e548502-52a1-49e1-8854-bc1d43931446.png)

Now, all guide lines are dimmed when you are inside nested statements:

![image](https://user-images.githubusercontent.com/85762931/220341161-d3fbc39b-71a6-4be8-81fb-086c1cc78976.png)

And only one guide line highlights if you select it:

![image](https://user-images.githubusercontent.com/85762931/220341440-84aee12b-ebbf-4216-a086-6ff733a974ce.png)

I don't know how I have the chance to see that feature on the first day of my usage but I hope you can make that a permanent option.

**Reply by izhangzhihao** [reply]
Please make sure your trial or paid license is still valid and choose this option: 
<img width="935" alt="image" src="https://user-images.githubusercontent.com/12044174/220493249-a8e2f9e6-5e33-4019-a304-00599684cb8e.png">

**Reply by BabakAmini** [reply]
I selected that option before but now "Only selected indent guide" option is selected. Maybe this happened when I disabled and re-enabled my plugins to investigate a bug in my editor. By the way, changing it resolved that issue for now however only part of the lines are painting now:

![image](https://user-images.githubusercontent.com/85762931/220536158-1342ae0f-da18-40fe-b0b7-e0927dcfc58d.png)

If you hover the mouse outside the editor or switch to another window the lines fully render. Then if you click on another place in the file this is what happens:

![image](https://user-images.githubusercontent.com/85762931/220537291-d86d12ca-2679-47e0-822d-e0cd1a8674d9.png)

**Reply by izhangzhihao** [reply]
Thanks for point this out, this is a known issue: https://github.com/izhangzhihao/intellij-rainbow-brackets/issues/2540

**Reply by izhangzhihao** [reply]
BTW, the latest release (just uploaded to JB marketplace) supports rainbow indent guidelines with-out highlight keywords now!
Please check the changelog here: https://github.com/izhangzhihao/intellij-rainbow-brackets/blob/2020.3/CHANGELOG.md#change-log

You may need 1 or 2 days to receive this update.

**Reply by BabakAmini** [reply]
I've updated to the latest version, and that option works for me, but I have a new issue.

Every time I restart my IDE, the "indent guide highlight option" changes back to "Only selected indent guide" and I have to set it back to the latest option in the list again. However, I think this is not related to the new update, and it was happening in the previous release too. Also, sometimes the license activation dialog box displays after IDE restarts and then the color generator for variables stops working even if it's checked in the settings. I have to uncheck and check it again and restart the IDE one more time to get them to work (if that dialog box doesn't display again).

**Reply by izhangzhihao** [reply]
Refer to https://github.com/izhangzhihao/intellij-rainbow-brackets/issues/2597#issuecomment-1445290853

**Comment by izhangzhihao** [comment]
I will convert this issue into a GitHub discussion. please continue to discuss your questions in the discussion.
